src/client/ui/cursor/index.js

```diff
@@ -1,97 +1,139 @@
+import hammerhead from './../deps/hammerhead';
 import testCafeCore from './../deps/testcafe-core';
-import CursorBehavior from './behavior';
-import CursorIFrameBehavior from './iframe-behavior';
+import CURSOR_UI_MESSAGES from './messages';
+
+
+var Promise        = hammerhead.Promise;
+var shadowUI       = hammerhead.shadowUI;
+var browserUtils   = hammerhead.utils.browser;
+var messageSandbox = hammerhead.eventSandbox.message;
+
+var styleUtils    = testCafeCore.styleUtils;
+var positionUtils = testCafeCore.positionUtils;
+var eventUtils    = testCafeCore.eventUtils;
+
+const CURSOR_CLASS       = 'cursor';
+const TOUCH_CLASS        = 'touch';
+const L_MOUSE_DOWN_CLASS = 'l-mouse-down';
+const R_MOUSE_DOWN_CLASS = 'r-mouse-down';
+const STATE_CLASSES      = [L_MOUSE_DOWN_CLASS, R_MOUSE_DOWN_CLASS].join(' ');
+
+var cursorElement  = null;
+var x              = 50;
+var y              = 50;
+var pointerOffsetX = 0;
+var pointerOffsetY = 0;
+
+// Setup cross-iframe interaction
+messageSandbox.on(messageSandbox.SERVICE_MSG_RECEIVED_EVENT, e => {
+    var msg = e.message;
+
+    switch (msg.cmd) {
+        case CURSOR_UI_MESSAGES.moveRequest:
+            var position = positionUtils.getIframePointRelativeToParentFrame({ x: msg.x, y: msg.y }, e.source);
+
+            CursorUI
+                .move(position.x, position.y)
+                .then(() => messageSandbox.sendServiceMsg({ cmd: CURSOR_UI_MESSAGES.moveResponse }, e.source));
+            break;
+
+        case CURSOR_UI_MESSAGES.leftButtonDownRequest:
+            CursorUI
+                .leftButtonDown()
+                .then(() => messageSandbox.sendServiceMsg({ cmd: CURSOR_UI_MESSAGES.leftButtonDownResponse }, e.source));
+            break;
+        case CURSOR_UI_MESSAGES.rightButtonDownRequest:
+            CursorUI
+                .rightButtonDown()
+                .then(() => messageSandbox.sendServiceMsg({ cmd: CURSOR_UI_MESSAGES.rightButtonDownResponse }, e.source));
+            break;
+        case CURSOR_UI_MESSAGES.buttonUpRequest:
+            CursorUI
+                .buttonUp()
+                .then(() => messageSandbox.sendServiceMsg({ cmd: CURSOR_UI_MESSAGES.buttonUpResponse }, e.source));
+            break;
+    }
+});
 
-var SETTINGS = testCafeCore.SETTINGS;
+export default class CursorUI {
+    static create () {
+        cursorElement = document.createElement('div');
```

Comment:
@inikulin It can't be a regular class, it's a singleton. It should be available when some messages from other frames received (sent by `iframeCursorUI`). So, I suppose we should export the object here.
In this case, `create` it's not a constructor but `private` function for deferred element creating. Maybe we should rename it to `createElement` or `createCursorElement`?